src/messaging/index.ts

```diff
@@ -0,0 +1,75 @@
+/*!
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FirebaseApp } from '../firebase-app';
+import * as messagingApi from './messaging';
+import * as messagingTypesApi from './messaging-types';
+import * as firebaseAdmin from '../index';
+
+export function messaging(app?: FirebaseApp): messagingApi.Messaging {
+  if (typeof(app) === 'undefined') {
+    app = firebaseAdmin.app();
+  }
+  return app.messaging();
+}
+
+/**
+ * We must define a namespace to make the typings work correctly. Otherwise
+ * `admin.messaging()` cannot be called like a function. Temporarily,
+ * admin.messaging is used as the namespace name because we cannot barrel 
+ * re-export the contents from messsaging, and we want it to
+ * match the namespacing in the re-export inside src/index.d.ts
+ */
+/* eslint-disable @typescript-eslint/no-namespace */
+export namespace admin.messaging {
+  // See https://github.com/microsoft/TypeScript/issues/4336
+  /* eslint-disable @typescript-eslint/no-unused-vars */
+  // See https://github.com/typescript-eslint/typescript-eslint/issues/363
+  export import AndroidConfig = messagingTypesApi.AndroidConfig;
```

Comment:
The current `index.d.ts` splits this based on legacy types, should I do that here?

https://github.com/firebase/firebase-admin-node/blob/1984f1fd592e3cddbc3c5b27655628dd878aaebc/src/index.d.ts#L779-L781

Reply:
Yes, that's a good idea.